pandas/tests/test_complex.py

```diff
@@ -0,0 +1,122 @@
+import numpy as np
+import pytest
+
+import pandas as pd
+from pandas import (
+    Index,
+    Series,
+    DataFrame,
+)
+import pandas.core.algorithms as algos
+import pandas.util.testing as tm
+
+
+class TestComplexSupportBasic:
+    @pytest.mark.parametrize("array,expected", [
+        (
+            [1 + 1j, 0, 1, 1j, 1 + 2j],
+            Series([1, 1, 1, 1, 1], index=[1 + 2j, 1 + 1j, 1j, 1, 0])
+        ),
+        (
+            [1 + 2j, 0, 1j, 1, 1j, 1 + 1j],
+            # index is sorted by value counts in descending order by default
+            Series([2, 1, 1, 1, 1], index=[1j, 1 + 2j, 1 + 1j, 1, 0])
+        )
+    ])
+    def test_value_counts(self, array, expected):
+        result = algos.value_counts(array)
+        tm.assert_series_equal(result, expected)
+
+    @pytest.mark.parametrize("array,expected", [
```

Comment:
For any that are just one set of parameters here you can just define in the function body. No need to parametrize

Reply:
This is just an initial set of test cases. I plan on adding more test cases for each of the algos.